db/migrate/20160321160404_add_middleware_event_streams.rb

```diff
@@ -0,0 +1,7 @@
+class AddMiddlewareEventStreams < ActiveRecord::Migration[5.0]
+  def change
+    add_column :event_streams, :middleware_type, :string   # relevant type (entity class name)
+    add_column :event_streams, :middleware_id, :bigint     # MIQ entity id
```

Comment:
Let's go with the hardcoded approach so that it's consistent with the other columns like vm and host.  In those particular cases, an event can refer to *both* a VM and a host at the same time, so we have both.  I can imagine that events could tie to a number of things, so if we are going to go generic, we're probably going to need to redesign the model altogether (with Cloud and Infra as wel)